client/src/app/core/services/csv-export.service.ts

```diff
@@ -0,0 +1,118 @@
+
+    /**
+     * Saves an array of model data to a CSV.
+     * @param data Array of Model instances to be saved
+     * @param columns Column definitions (model property; optional csv title)
+     * @param filename name of the resulting file
+     * @param options
+     */
+    public saveAsCsv(
+        data: BaseViewModel[],
+        columns: { property: string; title?: string }[],
+        filename: string,
+        { lineSeparator = '\r\n', columnSeparator = ';' }: { lineSeparator?: string; columnSeparator?: string }
+    ): boolean {
+        const all_lines = []; // Array of arrays of entries
+        const used_columns = []; // mapped properties to be included
+
+        // initial array of usable text separators. The first character not used
+        // in any text data or as column separator will be used as text separator
+        let ts_list = ['"', "'", '`', '/', '\\', ';', '.', ','];
+
+        if (lineSeparator === columnSeparator) {
+            throw new Error('TODO: lineseparator and columnseparator must differ from each other');
+        }
+
+        ts_list = this.checkCsvTextSafety(lineSeparator, ts_list);
+        ts_list = this.checkCsvTextSafety(columnSeparator, ts_list);
+
+        // create header data
+        const header = [];
+        for (const column of columns) {
+            if (column.title) {
+                ts_list = this.checkCsvTextSafety(column.title, ts_list);
+                header.push(column.title);
+            } else {
+                ts_list = this.checkCsvTextSafety(column.property, ts_list);
+                header.push(column.property);
+            }
+            used_columns.push(column.property);
+        }
+        all_lines.push(header);
+
+        // create lines
+        for (const item of data) {
+            const line = [];
+            for (const property of used_columns) {
+                ts_list = this.checkCsvTextSafety(item[property], ts_list);
+                line.push(item[property]);
+            }
+            all_lines.push(line);
+        }
+
+        // assemble lines, putting text separator in place
+
+        const final_lines = [];
+        if (ts_list.length) {
+            for (const line of all_lines) {
+                const newline = [];
+                for (const item of line) {
+                    if (typeof item === 'number') {
+                        newline.push(item);
+                    } else if (item === null || item === undefined || item === '') {
+                        newline.push('');
+                        // TODO export 0/1 for typeof === 'boolean' ?
+                    } else {
+                        const newstring = ts_list[0] + item + ts_list[0];
+                        newline.push(newstring);
+                    }
+                }
+                final_lines.push(newline.join(columnSeparator));
+            }
+            this.exporter.saveFile(final_lines.join(lineSeparator), filename);
+            return true;
```

Comment:
remove this and make the function return void. If somethings happens, an error can be thrown.